pkg/glanceapi/statefulset.go

```diff
@@ -290,19 +296,32 @@ func StatefulSet(
 		extraVolPropagation),
 		apiVolumes...)
 
-	// If possible two pods of the same service should not
-	// run on the same worker node. If this is not possible
-	// the get still created on the same worker node.
-	statefulset.Spec.Template.Spec.Affinity = affinity.DistributePods(
-		common.AppSelector,
-		[]string{
-			glance.ServiceName,
-		},
-		corev1.LabelHostname,
-	)
 	if instance.Spec.NodeSelector != nil {
 		statefulset.Spec.Template.Spec.NodeSelector = *instance.Spec.NodeSelector
 	}
 
+	if topology != nil {
+		ts := topology.Spec
+		if ts.TopologySpreadConstraint == nil {
+			ts.TopologySpreadConstraint = &tplg
+		} else {
+			statefulset.Spec.Template.Spec.TopologySpreadConstraints = *topology.Spec.TopologySpreadConstraint
+		}
+		if ts.Affinity != nil {
+			statefulset.Spec.Template.Spec.Affinity = ts.Affinity
+		}
+	} else {
```

Comment:
I like it much better to only apply the default antiaffinity if a topology was set. Merging them as in 587 could lead to unexpected behavior until the human operator figures out that the antiaffnity was getting merged.

Reply:
> I like it much better to only apply the default antiaffinity if a topology was set.

Not sure I understand. Are you saying that you prefer the previous behaviour?
Or is it just a typo and "only apply the default antiaffinity if a topology was *not* set"

Reply:
I think it's a typo. After some testing (we started in December) we had this feeling about always having `PodAntiAffinity` created by default might causes "conflicts/contradictions" in the scheduler and prevent the wanted behavior. I think this is the best approach, and both me and John are ok with that.